lib/plugins/physics.js

```diff
@@ -51,25 +53,36 @@ function inject (bot, { physicsEnabled }) {
 
   // This function should be executed each tick (every 0.05 seconds)
   // How it works: https://gafferongames.com/post/fix_your_timestep/
+
+  // WARNING: THIS IS NOT ACCURATE ON WINDOWS (15.6 Timer Resolution)
+  // use WSL or switch to Linux
+  // see: https://discord.com/channels/413438066984747026/519952494768685086/901948718255833158
   let timeAccumulator = 0
   function doPhysics () {
     const now = performance.now()
-    const deltaSeconds = (now - lastPhysicsFrameTime) / 1000
+    let deltaSeconds = (now - lastPhysicsFrameTime) / 1000
     lastPhysicsFrameTime = now
 
+    if (deltaSeconds >= 0.2) deltaSeconds = 0.2 // limiter (limit catchups to 4 ticks behind)
     timeAccumulator += deltaSeconds
 
     while (timeAccumulator >= PHYSICS_TIMESTEP) {
-      if (bot.physicsEnabled && shouldUsePhysics) {
-        physics.simulatePlayer(new PlayerState(bot, controlState), world).apply(bot)
-        bot.emit('physicsTick')
-        bot.emit('physicTick') // Deprecated, only exists to support old plugins. May be removed in the future
-      }
-      updatePosition(PHYSICS_TIMESTEP)
+      tickPhysics(now)
       timeAccumulator -= PHYSICS_TIMESTEP
     }
   }
 
+  function tickPhysics (now) {
+    if (bot.physicsEnabled && shouldUsePhysics) {
+      physics.simulatePlayer(new PlayerState(bot, controlState), world).apply(bot)
+      bot.emit('physicsTick')
+      bot.emit('physicTick') // Deprecated, only exists to support old plugins. May be removed in the future
```

Comment:
You could add a event deprecation warning here. Check if there are listeners on bot with the name physicTick with [this](https://nodejs.org/api/events.html#emitterlistenercounteventname) and if there are print a deprecation warning once.

Reply:
'physicTick event is deprecated! Please use physicsTick instead'
Is this a fine message?

Reply:
yes